flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamSort.scala

```diff
@@ -0,0 +1,169 @@
+import org.apache.flink.table.runtime.aggregate.SortUtil._
+
+/**
+  * Flink RelNode which matches along with Sort Rule.
+  *
+  */
+class DataStreamSort(
+  sort: LogicalSort,
+  cluster: RelOptCluster,
+  traitSet: RelTraitSet,
+  inputNode: RelNode,
+  rowRelDataType: RelDataType,
+  inputType: RelDataType,
+  description: String)
+    extends SingleRel(cluster, traitSet, inputNode) with DataStreamRel {
+
+  override def deriveRowType(): RelDataType = rowRelDataType
+
+  override def copy(traitSet: RelTraitSet, inputs: java.util.List[RelNode]): RelNode = {
+    new DataStreamSort(
+      sort,
+      cluster,
+      traitSet,
+      inputs.get(0),
+      rowRelDataType,
+      inputType,
+      description + sort.getId())
+  }
+
+  override def toString: String = {
+    s"Sort($sort)" +
+      " on fields: (${sort.collation.getFieldCollations})"
+  }
+
+  override def explainTerms(pw: RelWriter): RelWriter = {
+    super.explainTerms(pw)
+      .item("aggregate", sort)
+      .item("sort fields",sort.collation.getFieldCollations)
+      .itemIf("offset", sort.offset, sort.offset!=null)
+      .itemIf("fetch", sort.fetch, sort.fetch!=null)
+      .item("input", inputNode)
+  }
+
+  override def translateToPlan(tableEnv: StreamTableEnvironment): DataStream[Row] = {
+
+    val inputDS = getInput.asInstanceOf[DataStreamRel].translateToPlan(tableEnv)
+    
+    //need to identify time between others order fields. Time needs to be first sort element
+    val timeType = SortUtil.getTimeType(sort,inputType)
+    
+    //time ordering needs to be ascending
+    if (SortUtil.getTimeDirection(sort) != Direction.ASCENDING) {
+      throw new TableException("SQL/Table supports only ascending time ordering")
+    }
+      
+     
+    val (offset,fetch) = (sort.offset,sort.fetch)
+    
+    //enable to extend for other types of aggregates that will not be implemented in a window
+    timeType match {
+        case _: ProcTimeType =>
+            (offset,fetch) match {
+              case (o:Any,f:Any)  => null             // offset and fetch needs retraction
+              case (_,f:Any) => null                  // offset needs retraction
+              case (o:Any,_) => null                  // fetch needs retraction
+              case _ => createSortProcTime(inputDS)   //sort can be done with/without retraction
+            }
+        case _: RowTimeType =>
+          throw new TableException("SQL/Table does not support sort on row time")
+        case _ =>
+          throw new TableException("SQL/Table needs to have sort on time as first sort element")
+    }
+    
+  }
+
+  /**
+   * Create Sort logic based on processing time
+   */
+  def createSortProcTime(
+    inputDS: DataStream[Row]): DataStream[Row] = {
+
+          
+    // get the output types
+    //Sort does not do project.= Hence it will output also the ordering proctime field
+    //[TODO]Do we need to drop some of the ordering fields? (implement a projection logic?
```

Comment:
@fhueske i agree...but i did not see this in a proper plan in calcite. anyway..it might just be something temporal. I will drop the comment